YoutubeExplode/YoutubeClient.Static.cs

```diff
@@ -216,5 +216,38 @@ public static string ParseChannelId(string channelUrl)
                 ? result
                 : throw new FormatException($"Could not parse channel ID from given string [{channelUrl}].");
         }
+
+        /// <summary>
+        /// Tries to parses a username from a YouTube user URL
+        /// </summary>
+        public static bool TryParseUsername(string userUrl, out string username)
+        {
+            username = default(string);
+
+            if (userUrl.IsBlank())
+                return false;
+
+            var regularMatch = Regex.Match(userUrl, @"youtube\.com\/user\/([a-zA-Z0-9]{1,20})(?:\/|$)", RegexOptions.Multiline).Groups[1].Value;
```

Comment:
1) Why multiline?
2) I'd suggest to use `(.*?)(?:&|/|$)` instead of `([a-zA-Z0-9]{1,20})(?:\/|$)` just to make it simpler (there's no need to be very strict here), all validation logic should go to `ValidateUsername()` akin to `ValidateChannelId()`. Also remember to add `ValidateUsername()` to `GetChannelIdAsync()` method too later.

Reply:
It needs to be multiline to work, otherwise, links without "/" at the end would not match

I'll add what you've asked for

Reply:
Sorry, you're right. I checked with https://www.regextester.com/ and it showed no match unless I added the multiline flag, but I tried to execute it in c# and it seems to work even without it https://image.prntscr.com/image/W0BG11OiTEC0hb2l_fIGtQ.png

Reply:
I suggest using http://regexstorm.net/tester for testing .NET-flavored regex